### PR TITLE
Core/ts version task 410 presentation layer field validation and error handling

### PR DESCRIPTION
**Pull request by franken5tein**
## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/presentation/http/validators/http.validator.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            r.notEmpty().withMessage(`${ruleName} is required`),
+          );
+      }
+    });
+  }
+
+  static getFillByUser = [
+    ...this.getValidationRules(
+      ["userAddress", "network", "market", "start_time", "end_time"],
+      ["market", "start_time", "end_time"],
```
Comment (GreatGaryCooper) [inline ts-version/src/presentation/http/validators/http.validator.ts:62]: Can you tell me why market exists in both `ruleNames` and `optionalRules`?
Does putting something in the `optionalRules` override `ruleNames`?

If yes, pleas put a command above the `getValidationRules` function for future developers to not get confused. 🙏🏻

**Review by GreatGaryCooper — APPROVED** [review]

File: ts-version/src/presentation/http/validators/http.validator.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  ValidationChain,
+} from "express-validator";
+
+export class HTTPValidator {
+  static checkError(req: Request, res: Response, next: NextFunction) {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      throw new BadRequestError(
+        "Invalid request parameters",
+        errors.array().map((error) => `${error.param}, ${error.msg}`),
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply ts-version/src/presentation/http/validators/http.validator.ts:17]: Thank you for the positive feedback :)

File: ts-version/src/presentation/http/validators/http.validator.ts (hunk 1 above)
Comment (franken5tein) [reply ts-version/src/presentation/http/validators/http.validator.ts:62]: It's not, it just checks if the optionalRules is in the ruleNames and adds an optional check if so.

File: ts-version/src/presentation/http/validators/http.validator.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    start_time: (location: Location) => [location("start_time").isISO8601()],
+    end_time: (location: Location) => [location("end_time").isISO8601()],
+    limit: (location: Location) => [location("limit").isFloat({ min: 0 })],
+    fillId: (location: Location) => [location("fillId").isString()],
+    side: (location: Location) => [location("side").isIn(["buy", "sell"])],
+    signature: (location: Location) => [location("signature").isString()],
+    user_data: (location: Location) => [location("user_data").isBoolean()],
+    uuid: (location: Location) => [location("uuid").isUUID()],
+    tx: (location: Location) => [location("tx").isString()],
+    amount: (location: Location) => [location("amount").isFloat({ min: 0 })],
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply ts-version/src/presentation/http/validators/http.validator.ts:36]: Yes, it also accepts integers.

**Review by isaac-fibonacci — APPROVED** [review]

**Pull request merged**